File: src/test/resources/integration_tests/python/assertion_failed_integration_test.py

from org.vertx.testtools import VertxAssert
import vertx_tests
from core.event_bus import EventBus
import vertx

def test_fail_direct():
    VertxAssert.fail("failed")

def test_assert_failed_direct():
    VertxAssert.assertEquals("foo", "bar")

def test_fail_from_handler():
    @vertx.run_on_context
    def handler():
        VertxAssert.fail("failed")

def test_assert_failed_from_handler():
    @vertx.run_on_context
    def handler():
        VertxAssert.assertEquals("foo", "bar")

def test_fail_from_verticle():
    conf = {'type' : 'fail'}
    vertx.deploy_verticle('test_verticle.py', conf)

def test_assert_failed_from_verticle():
    conf = {'type' : 'assert_fail'}
    vertx.deploy_verticle('test_verticle.py', conf)


vertx_tests.start_tests(locals())

File: src/test/resources/integration_tests/javascript/assertionFailedIntegrationTest.js

var container = require("vertx/container")
var vertx = require("vertx");
var vertxTests = require("vertx_tests");
var vassert = require("vertx_assert");

function testFailDirect() {
  vassert.fail("failed")
}

function testAssertFailedDirect() {
  vassert.assertEquals("foo", "bar")
}

function testFailFromHandler() {
  vertx.runOnContext(function() {
    vassert.fail("failed")
  });
}

function testAssertFailedFromHandler() {
  vertx.runOnContext(function() {
    vassert.assertEquals("foo", "bar")
  });
}

function testFailFromVerticle() {
  container.deployVerticle("test_verticle.js", {type:"fail"})
}

function testAssertFailedFromVerticle() {
  container.deployVerticle("test_verticle.js", {type:"assert_fail"})
}

vertxTests.startTests(this);




